pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -294,38 +294,40 @@ func (s *Service) sendMsgEvents(msg *dispatcher.StateMsg) {
 func (s *Service) startInternalListener() {
 	go func() {
 		for msg := range s.callbackChannel {
-			// TODO handle error in callback - https://github.com/hyperledger/aries-framework-go/issues/242
-			if err := s.process(msg.ID, msg.DIDCommCallback); err != nil {
-				// TODO handle error
-				logger.Errorf(err.Error())
+			if !msg.DIDCommCallback.IsValid() {
+				continue
```

Comment:
This logic confuses me a bit. When will this scenario occur?

Reply:
For starters, it should be named `IsDone` or something similar.

Reply:
Why is there a msg in the queue that isn't "Done"?

Reply:
`why is the logic to simply skip it` it is according to previous logic (unit tests depend on it) it can happen when a client execute callback inappropriate way (should never happen) 
I would suggest here panic instead of `continue` because there is no another way to notify the client about the error or we can just log this error